### PR TITLE
[NPU] support model skywork-reward-gemma2-2-27B-v0.2

### PR DESCRIPTION
**Pull request by McZyWu**
## Motivation
support model skywork-reward-gemma2-2-27B-v0.2 

## Modifications
As follows.

## Accuracy Tests
when exporting SGLANG_NPU_FORWARD_NATIVE_GELUTANH=1 and SGLANG_NPU_FORWARD_NATIVE_GEMMA_RMS_NORM=1
accuracy is almost the same with hfrunner (GenericForSequenceClassification, Gemma2ForCausalLM) with tolerance under 4e-2. p.s. dtype should be same with model config (bfloat16). If you set dtype to float16, hf outputs will be nan.
image
image

### CODE REVIEW

File: python/sglang/srt/layers/layernorm.py [hunk 1]
```diff
@@ -455,6 +455,8 @@ def forward_npu(
         residual: Optional[torch.Tensor] = None,
         **kwargs,
     ) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
+        if get_bool_env_var("SGLANG_ENABLE_TORCH_COMPILE"):
```
Comment (Hexq0210) [inline python/sglang/srt/layers/layernorm.py]: dont use this env

Comment (McZyWu) [reply python/sglang/srt/layers/layernorm.py]: Adopted. Thank you.

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py [hunk 2]
```diff
@@ -223,6 +223,11 @@ def __init__(self, model_runner: ModelRunner):
             self.q_head_dim = self.qk_rope_head_dim + self.qk_nope_head_dim
         else:
             self.use_alibi = getattr(model_runner.model_config, "use_alibi", False)
+            if (
+                "Gemma2ForSequenceClassification" in
+                model_runner.model_config.hf_config.architectures
+            ):
+                self.use_native_sdpa = True
```
Comment (ping1jing2) [inline python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py:232]: please abstract a seperate func to handle this special cases, otherwise, this method will be huge

File: python/sglang/srt/layers/activation.py [hunk 3]
```diff
@@ -131,6 +132,8 @@ def forward_xpu(self, x: torch.Tensor) -> torch.Tensor:
         return self._forward_impl(x)
 
     def forward_npu(self, x: torch.Tensor) -> torch.Tensor:
+        if get_bool_env_var("FORWARD_NATIVE_GELUTANH"):
```
Comment (ping1jing2) [inline python/sglang/srt/layers/activation.py]: please use `sglang.srt.environ.Env.get()`

File: python/sglang/srt/layers/layernorm.py [hunk 4]
```diff
@@ -463,6 +463,8 @@ def forward_npu(
         residual: Optional[torch.Tensor] = None,
         post_residual_addition: Optional[torch.Tensor] = None,
     ) -> Union[torch.Tensor, Tuple[torch.Tensor, torch.Tensor]]:
+        if get_bool_env_var("FORWARD_NATIVE_GEMMA_RMS_NORM"):
```
Comment (ping1jing2) [inline python/sglang/srt/layers/layernorm.py]: ditto

File: test/registered/ascend/reward_models/test_ascend_gemma_2_27b_v0_2.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import os
+import unittest
+
+import torch
+
+from sglang.test.ci.ci_register import register_npu_ci
+from sglang.test.runners import HFRunner, SRTRunner
+from sglang.test.test_utils import CustomTestCase
+
+register_npu_ci(est_time=400, suite="nightly-1-npu-a3", nightly=True)
```
Comment (ping1jing2) [inline test/registered/ascend/reward_models/test_ascend_gemma_2_27b_v0_2.py:13]: i don't think we need to run this model in our nightly test

File: test/registered/ascend/reward_models/test_ascend_gemma_2_27b_v0_2.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        ) as srt_runner:
+            prompts = srt_runner.tokenizer.apply_chat_template(
+                convs, tokenize=False, return_dict=False
+            )
+            srt_outputs = srt_runner.forward(prompts)
+
+        hf_scores = torch.tensor(hf_outputs.scores)
+        srt_scores = torch.tensor(srt_outputs.scores)
+        print(f"{hf_scores=}")
+        print(f"{srt_scores=}")
```
Comment (ping1jing2) [inline test/registered/ascend/reward_models/test_ascend_gemma_2_27b_v0_2.py]: please use `logger` instead of `print`

File: python/sglang/srt/layers/activation.py (hunk 3 above)
Comment (McZyWu) [reply python/sglang/srt/layers/activation.py]: sure thanks

File: test/registered/ascend/reward_models/test_ascend_gemma_2_27b_v0_2.py (hunk 5 above)
Comment (McZyWu) [reply test/registered/ascend/reward_models/test_ascend_gemma_2_27b_v0_2.py:13]: Colleagues from the testing department think we need.

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py (hunk 2 above)
Comment (McZyWu) [reply python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py:232]: Please offer some feasible methods, thank u.

**Review by ping1jing2 — APPROVED** [review]

File: python/sglang/srt/environ.py [hunk 7]
```diff
@@ -463,6 +463,12 @@ class Envs:
     SGLANG_USE_AITER_FP8_PER_TOKEN = EnvBool(False)
     # fmt: on
 
+    # Forward native implementation for activation gelu tanh for model Skywork-Reward-Gemma-2-27B-v0.2
+    FORWARD_NATIVE_GELUTANH = EnvBool(False)
```
Comment (iforgetmyname) [inline python/sglang/srt/environ.py]: SGLANG_NPU_FORWARD_NATIVE_GELUTANH, and move to npu specific part

File: python/sglang/srt/environ.py [hunk 8]
```diff
@@ -463,6 +463,12 @@ class Envs:
...
+
+    # Forward native implementation for gemma rms norm for model Skywork-Reward-Gemma-2-27B-v0.2
+    FORWARD_NATIVE_GEMMA_RMS_NORM = EnvBool(False)
```
Comment (iforgetmyname) [inline python/sglang/srt/environ.py]: SGLANG_NPU_FORWARD_NATIVE_GEMMA_RMS_NORM

File: python/sglang/test/runners.py [hunk 9]
```diff
@@ -557,6 +557,7 @@ def __init__(
         json_model_override_args: Optional[dict[str, Any]] = None,
         lora_eviction_policy: str = "lru",
         enable_deterministic_inference: bool = False,
+        enable_torch_compile: bool = False,
```
Comment (iforgetmyname) [inline python/sglang/test/runners.py]: why we need this?

File: python/sglang/srt/environ.py (hunk 7 above)
Comment (McZyWu) [reply python/sglang/srt/environ.py]: sure. Thank u.

File: python/sglang/srt/environ.py (hunk 8 above)
Comment (McZyWu) [reply python/sglang/srt/environ.py]: sure. Thank u.

File: python/sglang/test/runners.py (hunk 9 above)
Comment (McZyWu) [reply python/sglang/test/runners.py]: This is because in the first version, I used this envs to control forward native function, and testing collegues wrote their scripts subsequently. However, as I switched to different envs seperately, this is not necessary, but we forgot to abandon this.
I've just removed this and tested. As it turns out, we don't need this.
